tests/v2/validation/provisioning/nodeproviders.go

```diff
@@ -33,20 +33,43 @@ func ExternalNodeProviderSetup(providerType string) ExternalNodeProvider {
 	case fromConfig:
 		return ExternalNodeProvider{
 			Name: providerType,
-			NodeCreationFunc: func(client *rancher.Client, numOfInstances int) ([]*nodes.Node, error) {
+			NodeCreationFunc: func(client *rancher.Client, numOfInstances int, numOfWinInstances int, multiconfig bool) (nodesList []*nodes.Node, winNodesList []*nodes.Node, err error) {
 				var nodeConfig nodes.ExternalNodeConfig
 				config.LoadConfig(nodes.ExternalNodeConfigConfigurationFileKey, &nodeConfig)
 
-				nodesList := nodeConfig.Nodes[numOfInstances]
-				for _, node := range nodesList {
-					sshKey, err := nodes.GetSSHKey(node.SSHKeyName)
-					if err != nil {
-						return nil, err
+				nodesList = nodeConfig.Nodes[numOfInstances]
+				winNodesList = nodeConfig.Nodes[numOfWinInstances]
+
+				if multiconfig {
+					for _, node := range nodesList {
+						sshKey, err := nodes.GetSSHKey(node.SSHKeyName)
+						if err != nil {
+							return nil, nil, err
+						}
+	
+						node.SSHKey = sshKey
 					}
+					for _, node2 := range winNodesList {
+						sshKey, err := nodes.GetSSHKey(node2.SSHKeyName)
+						if err != nil {
+							return nil, nil, err
+						}
 
-					node.SSHKey = sshKey
+						node2.SSHKey = sshKey
+					}
+				}
+				if !multiconfig {
```

Comment:
One little nitpick, this could be done with an else.` if () {}else{}`

Reply:
Done